src/transmission_manager.rs

```diff
@@ -106,37 +101,37 @@ fn schedule_next_infection_attempt(
             InfectiousStatus,
             InfectiousStatusType::Recovered,
         );
-    } else {
-        // Schedule the next infection attempt.
-        // Get the next infection attempt time.
-        let next_infection_times = get_next_infection_time(context, last_infection_time_uniform);
-
-        // Schedule the infection attempt. This function (a) grabs a contact at the time of the infection event
-        // to ensure that the contacts are current and (b) evaluates whether the transmission event is successful.
-        context.add_plan(
-            context.get_current_time() + next_infection_times.gi,
-            move |context| {
-                infection_attempt(context, transmitter_id)
-                    .expect("Error finding contact in infection attempt");
-                // Schedule the next infection attempt for this infected agent
-                // once the last infection attempt is over.
-                schedule_next_infection_attempt(
-                    context,
-                    transmitter_id,
-                    num_infection_attempts_remaining - 1,
-                    next_infection_times.uniform,
-                );
-            },
-        );
+        return;
     }
+
+    // Get the next infection attempt time.
+    let next_infection_times = get_next_infection_time(context, last_infection_time_uniform);
```

Comment:
If you instead do:

`let (next_infection_time_unif, next_infection_time_gi) = ....`

Then:

1. It will be clearer
2. You will not need to do .0 and .1
3. You will not need `move` on line 117

Reply:
Thanks! I didn't see this. Agree on improving clarity, but I can't get compilation without the `move` in the closure.